cupy/cuda/compiler.py

```diff
@@ -251,9 +267,18 @@ def _compile(
         else:
             headers = include_names = ()
 
+        arch_opt, method = _get_arch_for_options_for_nvrtc(arch, jitify)
```

Comment:
Is it ok to add `arch` even in HIP?

Reply:
I think this code is not executed for hip, but `compile_using_hiprtc`

Reply:
Seems this method is called from `_compile_with_cache_hip`: https://github.com/cupy/cupy/blob/8378fb776d98bce9f56d7aa0a6dc8e2b17631b66/cupy/cuda/compiler.py#L847-L852 😅

Reply:
yes, I got confused because there is both `compile_with_nvcc` and `compile_with_hipcc` :)
Should be fixed now :D